rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+        registry.apply_mutations_for_test(
+            vec![
+                dc_mutation("dc1"),
+                operator_mutation(operator(1), provider(1), 10, "dc1"),
+                operator_mutation(operator(2), provider(1), 10, "dc1"),
+                node_mutation(first_node.node_id(), operator(1)),
+            ]
+            .into_iter()
+            // These are the mutations required to have a compliant registry
+            .chain(get_mutations_to_achieve_invariancy(&[&first_node]))
+            .collect(),
+        );
+
+        let version_before = registry.latest_version();
+        registry
+            .do_replace_operator_(
+                payload(
+                    operator(1),
+                    operator(2),
+                    &[first_node.node_id(), first_node.node_id()],
+                ),
+                caller(1),
+            )
+            .assert_ok();
+
+        assert!(
+            registry.latest_version() == version_before + 1,
+            "Expected registry version to increase. Before execution: {}, after execution: {}",
+            version_before,
+            registry.latest_version()
+        );
+        for (operator, allowance) in &[(operator(1), 11), (operator(2), 9)] {
+            let operator_record = registry
+                .get(
+                    make_node_operator_record_key(*operator).as_bytes(),
+                    registry.latest_version(),
+                )
+                .unwrap();
+
+            let decoded = NodeOperatorRecord::decode(operator_record.value.as_slice()).unwrap();
+            assert_eq!(operator_record.version, version_before + 1);
+            assert_eq!(decoded.node_allowance, *allowance);
+        }
+    }
+
+    // Functions below are used to create an invariant
+    // registry. They are not important for the tests above
+    fn get_mutations_to_achieve_invariancy(
+        nodes: &[&ValidNodePublicKeys],
+    ) -> Vec<RegistryMutation> {
+        let one_more_nns_node = generate_valid_node_id();
+        let mut mutations = vec![
+            routing_table_mutation(),
+            subnet_list_record(),
+            nns(one_more_nns_node.node_id()),
+            replica_version_mutation(),
+            blessed_replica_versions_mutation(),
+            node_mutation(one_more_nns_node.node_id(), operator(150)),
+        ];
+
+        let threshold_pk_and_cup_mutations =
+            create_subnet_threshold_signing_pubkey_and_cup_mutations(
+                PrincipalId::new_subnet_test_id(0).into(),
+                &vec![(
+                    one_more_nns_node.node_id(),
+                    one_more_nns_node.dkg_dealing_encryption_key().clone(),
+                )]
+                .into_iter()
+                .collect(),
+            );
+        mutations.extend(threshold_pk_and_cup_mutations);
+
+        // One more node is required for
+        for node in nodes.iter().chain(&[&one_more_nns_node]) {
+            mutations.extend(vec![
```

Comment:
There's a function for getting mutations to add nodes called "make_add_node_registry_mutations".  There's also a function "new_node_keys_and_node_id" that generates keys and IDs that go together.  See `prepare_registry_with_nodes_and_node_operator_id` for an example.

We shouldn't recreate all of these here, but rather add to the test_utils and update the existing test utils to use more generic functions (if we need the more generic functions).

Reply:
I've refactored the code to use the helper functions that already exist, the missing piece was `prepare_registry_with_nodes_and_node_operator_id` so all of the excess functions have been removed.